arviz/plots/ecdfplot.py

```diff
@@ -0,0 +1,341 @@
+        x_coord=x_coord,
+        y_coord=y_coord,
+        x_bands=x,
+        lower=lower,
+        higher=higher,
+        confidence_bands=confidence_bands,
+        figsize=figsize,
+        fill_band=fill_band,
+        plot_kwargs=plot_kwargs,
+        fill_kwargs=fill_kwargs,
+        plot_outline_kwargs=plot_outline_kwargs,
+        ax=ax,
+        show=show,
+        backend_kwargs=backend_kwargs,
+        **kwargs
+    )
+
+    if backend is None:
+        backend = rcParams["plot.backend"]
+    backend = backend.lower()
+
+    plot = get_plotting_function("plot_ecdf", "ecdfplot", backend)
+    ax = plot(**ecdf_plot_args)
+
+    return ax
+
+
+def compute_ecdf(sample, z):
+    """Compute ECDF.
+
+    This function computes the ecdf value at the evaluation point
+        or a sorted set of evaluation points.
+    """
+    if not isinstance(z, np.ndarray):
+        ## if z is just an instance then use Binary search
+        left, right = 0, len(sample) - 1
+        while left <= right:
+            mid = int((left + right) / 2)
+            if sample[mid] > z:
+                right = mid - 1
+            else:
+                left = mid + 1
+        return left / len(sample)
+    else:
+        ## if z is a list then follow this approach
+        f_z = np.empty(len(z))
+        u_idx = 0
+        for i, z_i in enumerate(z):
+            while u_idx < len(sample) and sample[u_idx] < z_i:
+                u_idx += 1
+            f_z[i] = u_idx + 1
+
+        return f_z / len(sample)
+
+
+def get_ecdf_points(x, probs, difference):
+    """Compute the coordinates for the ecdf points using compute_ecdf."""
+    if difference:
+        x_coord, y_coord = np.empty(len(x)), np.empty(len(x))
+    else:
+        x_coord, y_coord = np.empty(len(x) + 1), np.empty(len(x) + 1)
+        ## pseudo point at the start of ecdf plot so that it touches x-axis
+        x_coord[0], y_coord[0] = x[0], 0
+
+    for i, x_i in enumerate(x):
+        f_x_i = compute_ecdf(probs, x_i)
+        if difference:
+            x_coord[i], y_coord[i] = x_i, f_x_i
+        else:
+            x_coord[i + 1], y_coord[i + 1] = x_i, f_x_i
+    return x_coord, y_coord
+
+
+def compute_gamma(n, z, npoints=None, num_trials=1000, fpr=0.05):
+    """Compute gamma for confidence interval calculation.
+
+    This function simulates an adjusted value of gamma to account for multiplicity
+    when forming an 1-fpr level confidence envelope for the ECDF of a sample.
+    """
+    if npoints is None:
+        npoints = n
+    gamma = []
+    for _ in range(num_trials):
+        unif_samples = uniform.rvs(0, 1, n)
+        unif_samples = np.sort(unif_samples)
+        gamma_m = 1000
+        ## Can compute ecdf for all the z together or one at a time.
+        f_z = compute_ecdf(unif_samples, z)
+        for i in range(npoints):
+            curr = min(binom.cdf(n * f_z[i], n, z[i]), 1 - binom.cdf(n * f_z[i] - 1, n, z[i]))
+            gamma_m = min(2 * curr, gamma_m)
```

Comment:
This vectorized version should be much faster
```suggestion
        f_z = compute_ecdf(unif_samples, z)
        gamma_m = 2 * min(np.amin(binom.cdf(n * f_z, n, z)), np.amin(1 - binom.cdf(n * f_z - 1, n, z)))
```